lib/registries/addon/helpers/math.ts

```diff
@@ -0,0 +1,96 @@
+import Helper from '@ember/component/helper';
+import { htmlSafe } from '@ember/string';
+import katex, { KatexRenderOptions } from 'katex';
+
+export interface Delimiter {
+    start: string;
+    end: string;
+    inline: boolean;
+}
+
+interface Match {
+    start: number;
+    end: number;
+    delimiter: Delimiter;
+}
+
+// Adapted from https://github.com/Khan/KaTeX/blob/master/contrib/auto-render/splitAtDelimiters.js
+export function findEnd(content: string, start: number, end: string) {
+    let contextLevel = 0;
+
+    for (let i = start; i < content.length; i++) {
+        if (contextLevel < 1 && content.startsWith(end, i)) {
+            return i + end.length;
+        } else if (content[i] === '}') {
+            contextLevel++;
+        } else if (content[i] === '{') {
+            contextLevel--;
+        } else if (content[i] === '\\') {
+            i++;
+        }
+    }
+
+    return -1;
+}
+
+export function replace(content: string, delimiters: Delimiter[], callback: (exp: string, delim: Delimiter) => string) {
+    const matches: Match[] = [];
+
+    for (let i = 0; i < content.length; i++) {
+        for (const delimiter of delimiters) {
+            if (content.startsWith(delimiter.start, i)) {
+                const start = i;
+                const end = findEnd(content, i + delimiter.start.length, delimiter.end);
+
+                if (end === -1) {
+                    continue;
+                }
+
+                i = end;
+                matches.push({ end, start, delimiter });
+            }
+        }
+    }
+
+    let index = 0;
+    let result = '';
+
+    for (const match of matches) {
+        result += content.slice(index, match.start);
+        result += callback(content.slice(
+            match.start + match.delimiter.start.length,
+            match.end - match.delimiter.end.length,
+        ), match.delimiter);
+
+        index = match.end;
+    }
+
+    return result + content.slice(index);
+}
+
+export const DELIMITERS: Delimiter[] = [
+    { start: '$$', end: '$$', inline: false },
+    { start: '\\[', end: '\\]', inline: false },
+    { start: '$', end: '$', inline: true },
```

Comment:
This one seems easy to hit accidentally, like "The GDP of Lemonadia increased from $`7 all the way to `$12 in one week." Does normal text rendered as math look weird?

Reply:
It will look weird but I don't actually think it will be that common to hit. However using `$ expr $` seem to be fairly standard?

Currently MathJax will mess it up as well and we're [explicitly opting into that behavior](https://github.com/CenterForOpenScience/ember-osf-preprints/blob/develop/app/index.html#L19).

![screen shot 2018-10-04 at 10 29 29 am](https://user-images.githubusercontent.com/5532905/46481144-b50d5c00-c7c0-11e8-9444-79a1606221d7.png)

Reply:
Hm, would it be reasonable to put spaces in the delimiter? Maybe `start: ' $ ', end: ' $ '` or `start: '$ ', end: ' $'`? I guess it feels like using a dollar symbol twice in one title or description might happen pretty often in some fields.